### PR TITLE
feat(sandbox): make icons searchable by keyword

### PR DESCRIPTION
**Pull request by mihai-peteu**
# Summary
Screenshot 2024-06-19 at 09 12 35

### CODE REVIEW

File: sandbox/pages/HomePage.vue [hunk 1]
```diff
@@ -81,23 +83,45 @@ const filteredComponents = computed(() => {
       type: 'multi-color',
       name: key,
       component: val,
+      keyword: '',
```
Comment (portikM) [inline sandbox/pages/HomePage.vue:86]: I suggest we name it plural (`keywords`) and make it an array. Some other icons could benefit from having multiple search terms defined as keywoards (e.g. CogIcon -> `keywoards: ['gear', 'settings']`). I don't ask you to add keywords to that icon - I just use it as an example as to why array would be more beneficial

File: sandbox/pages/HomePage.vue [hunk 2]
```diff
@@ -81,23 +83,45 @@ const filteredComponents = computed(() => {
...
   if (!searchQuery.value || searchQuery.value?.toLowerCase() === 'icon') {
     return allComponents
   }
 
-  return allComponents.filter((icon: any) => icon.name.toLowerCase().includes(searchQuery.value.toLowerCase().replace(/icon/gi, '')))
+  const searchTerm = searchQuery.value.toLowerCase().replace(/icon/gi, '')
+
+  return allComponents.filter((icon: any) => {
+    return icon.name.toLowerCase().includes(searchTerm) ||
+      icon?.keyword?.toLowerCase().includes(searchTerm)
```
Comment (portikM) [inline sandbox/pages/HomePage.vue:123]: If you make `keywords` an array you will have to remove `.toLowerCase()` from here but I suggest you transform all keywords to lowercase when you're adding them to the `keywords` array instead (line 111)

File: sandbox/pages/HomePage.vue [hunk 3]
```diff
@@ -81,23 +83,45 @@ const filteredComponents = computed(() => {
...
+
+    // Grab 2-letter country code from icon name
+    const match = /Flag(.*?)Icon/.exec(key) || ''
+    const countryCode = match[1].toUpperCase()
+
     allComponents.push({
       type: 'flags',
       name: key,
       component: val,
+      keyword: countryMap.has(countryCode) ? countryMap.get(countryCode)?.name : '',
```
[the remark this answers is not in the document]
Comment (mihai-peteu) [reply sandbox/pages/HomePage.vue]: Linter doesn't like it.  A JS Map's `.get()` can still return undefined....

https://www.basedash.com/blog/why-typescript-map-get-returns-undefined#:~:text=USING%20OPTIONAL%20CHAINING

**Review by portikM — DISMISSED** [review]

**Review by adamdehaven — CHANGES_REQUESTED** [review]
I think we’re going to need to make some adjustments to lazy load the icons or something; loading the sandbox takes upwards of 3-4 seconds with all of the flag icons

the search is super-laggy too and needs to be debounced

**Comment by adamdehaven** [comment]
the lag seems to be related to search; the current site loads pretty quickly

**Comment by mihai-peteu** [comment]
@adamdehaven the search input was already debounced, i lowered the timeout from 500ms to 200ms
https://github.com/Kong/icons/pull/322/commits/d58093fe7bb887bdd7e324b97ca819b8b8d290ea#diff-f5b67e1dd2f5e869bf6f35aadae0f59bd40f4f96aaebb91de2b99876e57740e2R50

If you look at my last commit, I think maybe the other reason for the perceived slowness could have been that were were redefining `allComponents` inside of `filteredComponents` every time the search value was changed 😬 

Should be considerably more performant now.

**Review by adamdehaven — COMMENTED** [review]
The lag is much less noticeable with your latest commit. 

Would it make sense to display the country names under the Flag icon names in the card? Might make it easier to identify the flag you're looking for when searching.

**Comment by mihai-peteu** [comment]
Updated:

Screenshot 2024-06-19 at 09 12 35

**Review by adamdehaven — APPROVED** [review]

**Pull request merged**